### PR TITLE
Feature/digital green certificate

### PR DESCRIPTION
**Pull request by AliceBilc**
## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] Make sure `mvn -P integration-tests clean verify` runs for the whole project and, if you touched any code for a service in the `services` folder, ensure it can be run with `spring-boot:run`

## Description

Place copy of the json files in our code-line/image and generate the 'en' protobuf + signature. 
https://jira-ibs.wbs.net.sap/browse/EXPOSUREBACK-1202

### CODE REVIEW

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/qrcode/QrCodeTemplateLoader.java [hunk 1]
```diff
@@ -1,42 +1,46 @@
...
 import org.springframework.stereotype.Component;
 
 @Component
 public class QrCodeTemplateLoader {
 
   private static final Logger logger = LoggerFactory.getLogger(QrCodeTemplateLoader.class);
 
-  private final DistributionServiceConfig config;
+  @Autowired
+  DistributionServiceConfig config;
```
Comment (roesslerj) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/qrcode/QrCodeTemplateLoader.java:21]: These changes are from me and shoulnd't have creeped in here. I'll remove them for a separate PR...

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/component/DigitalGreenCertificateStructureProvider.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+  /**
+   * Returns the publishable archive with the Digital Green Certificates protobuf structure for mobile clients.
+   */
+  public DirectoryOnDisk getDigitalGreenCertificates() {
+    return constructArchiveToPublish(distributionServiceConfig.getDigitalGreenCertificate(),
+        dgcToProtobufMapping.constructProtobufMapping());
+  }
+
+  private DirectoryOnDisk constructArchiveToPublish(
```
Comment (hilmarf) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/component/DigitalGreenCertificateStructureProvider.java:46]: needs to be enhanced with further development by additional languages

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/DigitalGreenCertificateToProtobufMapping.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        logger.error("Error reading {} from json {}.", rawType.getSimpleName(), path, e);
+      }
+    }
+    try (InputStream jsonStream = resourceLoader.getResource(defaultPath).getInputStream()) {
+      // fallback to default
+      return deserializeJsonToSimpleType(jsonStream, rawType);
+    } catch (IOException e) {
+      logger.error("We could not load the default {}. This shouldn't happen!", defaultPath, e);
+      throw new RuntimeException("We could not load the default " + defaultPath
+          + ". This shouldn't happen!", e);
```
Comment (hilmarf) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/DigitalGreenCertificateToProtobufMapping.java:104]: @roesslerj / @AliceBilc 
will this abort the distribution of keys or event-checkins?

File: services/distribution/src/main/resources/application.yaml [hunk 4]
```diff
@@ -199,6 +199,12 @@ services:
       plausible-deniability-parameters:
         probability-to-fake-check-ins-if-no-check-ins: ${EVREG_PROBABILITY_TO_FAKE_CHECK_INS_IF_NO_CHECK_INS:0}
         probability-to-fake-check-ins-if-some-check-ins: ${EVREG_PROBABILITY_TO_FAKE_CHECK_INS_IF_SOME_CHECK_INS:0.005}
+    digital-green-certificate:
+      mah-json-path: ${DIGITAL_GREEN_CERTIFICATE_MAH_JSON_PATH:}
+      prophylaxis-json-path: ${DIGITAL_GREEN_CERTIFICATE_PROPHYLAXIS_JSON_PATH:}
+      medicinal-products-json-path: ${DIGITAL_GREEN_CERTIFICATE_MEDICINAL_PRODUCTS_JSON_PATH:}
```
Comment (hilmarf) [inline services/distribution/src/main/resources/application.yaml:205]: don't we have proper defaults?

**Review by hilmarf — APPROVED** [review]

**Review by roesslerj — APPROVED** [review]

File: services/distribution/src/main/resources/application.yaml (hunk 4 above)
Comment (roesslerj) [reply services/distribution/src/main/resources/application.yaml:205]: Our defaults/fallbacks are hard-coded, so we don't want them configured here.

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/DigitalGreenCertificateToProtobufMapping.java (hunk 3 above)
Comment (roesslerj) [reply services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/DigitalGreenCertificateToProtobufMapping.java:104]: Will be addressed in future PR (already implemented)

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/component/DigitalGreenCertificateStructureProvider.java (hunk 2 above)
Comment (roesslerj) [reply services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/component/DigitalGreenCertificateStructureProvider.java:46]: Being implemented right now and is documented by EXPOSUREBACK-1270

**Pull request merged**